Integration group by query test no time values in timestamp field error

```
--- FAIL: TestRawQueryGroupBy (7.78s)
    raw_query_groupby_test.go:46:
                Error Trace:    raw_query_groupby_test.go:46
                Error:          Received unexpected error:
                                received '500' status code: {"state":"Error","error":"error performing grouped query {testNamespace 1548115200000000000 1548201600000000000 [] [[st] [tt]] [] [] 10 0x43d21d0 map[12889413958949747480:{[@timestamp] map[0:map[time:{}]]} 10496096952681553436:{[st] map[1:map[time:{} null:{} bool:{} int:{} double:{} string:{}]]} 1294810295590724393:{[tt] map[2:map[int:{} double:{} string:{} time:{} null:{} bool:{}]]}]} against database namespace testNamespace: no time values in timestamp field"}
                Test:           TestRawQueryGroupBy
```

There seems to be a race condition w/ the writes not going through before the query hits.

--- a/storage/immutable_segment.go
+++ b/storage/immutable_segment.go
@@ -196,6 +196,10 @@ func (s *immutableSeg) QueryRaw(
 		return nil
 	}
 
+	if s.NumDocuments() == 0 {
+		return nil
+	}
+
 	// Check if this segment has all orderBy fields as they are required, and bail early
 	// otherwise.
 	for _, ob := range q.OrderBy {
@@ -285,6 +289,10 @@ func (s *immutableSeg) QueryGrouped(
 		return nil
 	}
 
+	if s.NumDocuments() == 0 {
+		return nil
+	}
+
 	// Check if this segment has all groupBy fields as they are required, and bail early
 	// otherwise.
 	for _, gb := range q.GroupBy {
@@ -370,6 +378,10 @@ func (s *immutableSeg) QueryTimeBucket(
 	q query.ParsedTimeBucketQuery,
 	res *query.TimeBucketResults,
 ) error {
+	if s.NumDocuments() == 0 {
+		return nil
+	}
+
 	// Identify the set of fields needed for query execution.
 	allowedFieldTypes, fieldIndexMap, fieldsToRetrieve, err := s.collectFieldsForQuery(
 		q.NumFieldsForQuery(),

--- a/storage/mutable_segment.go
+++ b/storage/mutable_segment.go
@@ -270,6 +270,12 @@ func (s *mutableSeg) QueryGrouped(
 		return nil, errMutableSegmentAlreadySealed
 	}
 
+	numDocuments := s.mutableSegmentBase.NumDocuments()
+	if numDocuments == 0 {
+		s.RUnlock()
+		return q.NewGroupedResults(), nil
+	}
+
 	allowedFieldTypes, fieldIndexMap, queryFields, err := s.collectFieldsForQueryWithLock(
 		q.NumFieldsForQuery(),
 		q.FieldConstraints,
@@ -278,8 +284,6 @@ func (s *mutableSeg) QueryGrouped(
 		s.RUnlock()
 		return nil, err
 	}
-
-	numDocuments := s.mutableSegmentBase.NumDocuments()
 	s.RUnlock()
 
 	defer func() {

